src/Core/Parser/Sequencer.php

```diff
@@ -486,6 +501,36 @@ protected function sequenceTagNode(): ?ComponentInterface
                         return null;
                     }
 
+                    $argumentValidationError = null;
+                    $nodeHash = spl_object_hash($viewHelperNode);
+                    // Check if we already have a validation error on the sequencing error stack
+                    foreach ($this->nodeSequencingExceptionStack[$nodeHash] ?? [] as $sequencingException) {
+                        if ($sequencingException->getCode() === 1621337984) {
+                            $argumentValidationError = $sequencingException;
+                            break;
+                        }
+                    }
+
+                    try {
```

Comment:
This try/catch, including the throw, should only happen if `$argumentValidationError !== null` since we don't want to re-validate arguments on every valid node - only those that previously added an error to the stack.

Then you can also skip the condition below and directly throw the error:

```php
throw ($argumentValidationError ?? $this->createErrorAtPosition(...));
```

...and better resolve the exception code from `$e->getCode()` instead of hardcoding it here.